maptivist/src/main/webapp/script.js

```diff
@@ -12,27 +12,149 @@
+        return;
+        }
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+    // If the place has a geometry, then present it on a map.
+    if (place.geometry.viewport) {
+        map.fitBounds(place.geometry.viewport);
+    } else {
+        map.setCenter(place.geometry.location);
+        map.setZoom(17);  
+    }
+    
+    // Concatentates a readable address for an autocomplete result
+    var address = '';
+        if (place.address_components) {
+            address = [
+                (place.address_components[0] && place.address_components[0].short_name || ''),
+                (place.address_components[1] && place.address_components[1].short_name || ''),
+                (place.address_components[2] && place.address_components[2].short_name || '')
+            ].join(' ');
+        }
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+        // Add the result contents to the infowindow display
+        infowindowContent.children['place-icon'].src = place.icon;
+        infowindowContent.children['place-name'].textContent = place.name;
+        infowindowContent.children['place-address'].textContent = address;
+        infowindow.open(map);
+  }); 
+}
+
+/** Adds a new Marker based on form submission
+ */
+
+function createMarker() {
+  // Get the coordinates from the form input and create a new Latlng object
+  var latitude = parseFloat(document.getElementById('marker-lat').value);
+  var longitude = parseFloat(document.getElementById('marker-lng').value);
+  var myLatlng = new google.maps.LatLng(latitude, longitude);
+  
+  // Create a new marker, it assumed that the position is a private attribute that cannot be accessed
+  var marker = new google.maps.Marker({
+    position: myLatlng,
+    map: map,
+    title: document.getElementById('marker-title').value    
+  });
+
+  // Adds the new marker to the map and pans to the marker 
+  marker.setMap(map);
+  map.panTo(marker.getPosition());
+
+  // Adds the new infowindow to the marker
+  var infowindow = createInfowindow(marker.getPosition());
+  marker.addListener('click', function() {
+    infowindow.open(map, marker);
+  });
 }
 
 /**
-* Intializes primary map for the main webpage
+* Adds an infowindow based on the marker creation form
  */
-function initMap(){}
+function createInfowindow(position) {
+  //Set info window content from form
+  var title = document.getElementById('marker-title').value;
+  var location = position.toString();
+  var description = document.getElementById('marker-description').value;
+  var link = document.getElementById('marker-link').value;
+  var category = document.getElementsByClassName('marker-category').value;
+  
+  // Set the content of the info window 
+  var contentString = `<div id="marker-window"><h1>${title}</h1><br><h2>${location}</h2><br><p>${description}</p><br><p>${link}</p><br>`
```

Comment:
I think template literals (the ``) allow newlines. Could you format the HTML to be more readable with newlines? (e.g. after all of the closing tags?

Reply:
Ok sure I'll try that!